drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -643,26 +716,60 @@ static void _isr(netdev_t *netdev)
     }
 
     /* read (consume) device status */
-#if defined(MODULE_AT86RFA1) || defined(MODULE_AT86RFR2)
-    irq_mask = dev->irq_status;
-    dev->irq_status = 0;
-#else
-    irq_mask = at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+    switch (dev->base.dev_type) {
+        default:
+            irq_mask = at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+            break;
+#if IS_USED(MODULE_AT86RFA1)
+        case AT86RF2XX_DEV_TYPE_AT86RFA1: {
+            irq_mask = ((at86rfa1_t *)dev)->irq_status;
+            ((at86rfa1_t *)dev)->irq_status = 0;
+            break;
+        }
+#endif
+#if IS_USED(MODULE_AT86RFR2)
+        case AT86RF2XX_DEV_TYPE_AT86RFR2: {
+            irq_mask = ((at86rfr2_t *)dev)->irq_status;
+            ((at86rfr2_t *)dev)->irq_status = 0;
+            break;
+        }
 #endif
```

Comment:
I don't think it's useful to duplicate the code for those two.
You could also have a `static inline _get_irq_status()` function to get the `#ifdef` out of  `_isr()`

Reply:
Ok, I can do a `static inline static inline _get_irq_status()`.

Unfortunately, I cannot do:
```C
#if IS_USED(MODULE_AT86RFA1) || IS_USED(MODULE_AT86RFR2)
case AT86RF2XX_DEV_TYPE_AT86RFA1:
case AT86RF2XX_DEV_TYPE_AT86RFR2: {
 /* code */
}
#endif
```

because the `AT86RF2XX_DEV_TYPE_*` is only known to the linker if the module is included.
But that´s part of the concept to reduce memory.

If I always have defined all device types, I always must have all values in `at86rf2xx_properties.c` defined to keep the index base property access working. And it would need more RAM. If the compiler sees that the program does the exact same thing for two cases, I thought it gets optimized to one case?

Do you prefer to have all types always available and for that increase RAM size?

Reply:
Hm I see. Of course I'd rather save on ~~RAM~~ ROM (those arrays are `const` so they'll reside in ROM).

>  If the compiler sees that the program does the exact same thing for two cases, I thought it gets optimized to one case?

I know the compiler does this for strings, but for code I wouldn't be so sure.
Anyway it doesn't matter in this case as you can't have both AT86RFA1 & AT86RFR2 enabled.

Reply:
On AVR they will end up consuming both RAM and ROM :-(

In this regard  the von-Neumann architecture simply is better.